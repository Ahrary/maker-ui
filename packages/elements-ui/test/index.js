import React from 'react'
import { Box } from 'theme-ui'
import renderer from 'react-test-renderer'
import { cleanup } from '@testing-library/react'
import { matchers } from 'jest-emotion'
import {
  Layout,
  Topbar,
  Header,
  Navbar,
  MobileMenu,
  Content,
  Main,
  SideNav,
  Sidebar,
  Footer,
  Section,
} from '../src/components'

afterEach(cleanup)

expect.extend(matchers)

const renderJSON = e => renderer.create(e).toJSON()

const theme = {
  colors: {
    bg_header: '#000',
  },
}

const options = { navigation: 'sidebar-content' }

describe('Layout', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <h1>Test</h1>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })

  test('user theme overrides default values', () => {
    const json = renderJSON(
      <Layout theme={theme}>
        <Box sx={{ bg: 'bg_header' }}></Box>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })

  test('user options override default options', () => {
    const json = renderJSON(
      <Layout options={{ layout: 'content-sidebar' }}>
        <Content />
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Topbar', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Topbar>Test</Topbar>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Header', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Header>Test</Header>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Navbar', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Header>
          <Navbar />
        </Header>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('MobileMenu', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Header>
          <MobileMenu />
        </Header>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Content', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Content />
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Main', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Content>
          <Main>Test</Main>
        </Content>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})
describe('Sidebar', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Content>
          <Sidebar />
          <Main />
        </Content>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})
describe('SideNav', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <SideNav />
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Footer', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Footer />
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Section', () => {
  test('renders', () => {
    const json = renderJSON(
      <Layout>
        <Section>Test</Section>
      </Layout>
    )
    expect(json).toMatchSnapshot()
  })
})
