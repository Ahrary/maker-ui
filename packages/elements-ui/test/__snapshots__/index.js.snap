// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
}

<div
  className="emotion-2"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <div
    className="emotion-1"
  />
</div>
`;

exports[`Footer renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_footer,#fff);
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <footer
    className="emotion-2"
    id="footer"
    role="contentinfo"
    tabIndex="-1"
  >
    <div
      className="emotion-1"
    />
  </footer>
</div>
`;

exports[`Header renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 100;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: var(--theme-ui-colors-bg_header,#fff);
}

<div
  className="emotion-2"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <header
    className="emotion-1"
    id="site-header"
    role="banner"
  >
    Test
  </header>
</div>
`;

exports[`Layout renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

<div
  className="emotion-1"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <h1>
    Test
  </h1>
</div>
`;

exports[`Layout user theme overrides default values 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#000);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_header,#000);
}

<div
  className="emotion-2"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <div
    className="emotion-1"
  />
</div>
`;

exports[`Main renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <div
    className="emotion-2"
  >
    <main
      className="emotion-1"
      id="content"
      role="main"
    >
      Test
    </main>
  </div>
</div>
`;

exports[`MobileMenu renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 100;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: var(--theme-ui-colors-bg_header,#fff);
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.2);
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all ease .4s;
  transition: all ease .4s;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  background-color: var(--theme-ui-colors-bg_mobileMenu,rgba(0,0,0,0.9));
  top: 0;
  bottom: 0;
  -webkit-transition: all ease .3s;
  transition: all ease .3s;
  opacity: 1;
  visibility: hidden;
  left: 0;
  width: 300px;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

@media screen and (min-width:750px) {
  .emotion-2 {
    width: 30vw;
  }
}

<div
  className="emotion-4"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <header
    className="emotion-3"
    id="site-header"
    role="banner"
  >
    <div
      className="emotion-1"
      id="menu-overlay"
      onClick={[Function]}
      role="presentation"
    />
    <div
      className="emotion-2"
      id="mobile-nav"
    />
  </header>
</div>
`;

exports[`Navbar renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 100;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: var(--theme-ui-colors-bg_header,#fff);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  margin: 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1460px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
}

@media screen and (min-width:750px) {
  .emotion-3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4px;
  width: 32px;
  height: 32px;
  color: inherit;
  background-color: transparent;
  border: none;
  border-radius: 4px;
  display: block;
}

@media screen and (min-width:750px) {
  .emotion-5 {
    display: none;
  }
}

.emotion-5 svg {
  margin: 0 auto;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  margin: 0;
}

<div
  className="emotion-10"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <header
    className="emotion-9"
    id="site-header"
    role="banner"
  >
    <div
      className="emotion-8"
    >
      <div
        className="emotion-1"
        id="site-logo"
      >
        <a
          href="/"
        >
          Logo
        </a>
      </div>
      <div
        className="emotion-6"
      >
        <nav
          className="nav-primary emotion-3"
        >
          <ul
            className="menu-primary emotion-1"
          />
        </nav>
        <div
          className="emotion-6"
        >
          <button
            aria-expanded="false"
            aria-label="Toggle Menu"
            className="emotion-5"
            onClick={[Function]}
            title="Menu"
          >
            <svg
              className="emotion-4"
              fill="currentcolor"
              height={24}
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Section renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <section
    className="emotion-2"
  >
    <div
      className="emotion-1"
    >
      Test
    </div>
  </section>
</div>
`;

exports[`SideNav renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_sideNav,#333);
  width: 300px;
  position: fixed;
  top: 0;
  bottom: 0;
  z-index: 100;
  -webkit-transform: translateX(300px);
  -ms-transform: translateX(300px);
  transform: translateX(300px);
  -webkit-transition: -webkit-transform ease .3s;
  -webkit-transition: transform ease .3s;
  transition: transform ease .3s;
}

@media screen and (min-width:750px) {
  .emotion-1 {
    position: relative;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  display: inline-block;
  bottom: 30px;
}

@media screen and (min-width:750px) {
  .emotion-2 {
    display: none;
  }
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <div
    className="emotion-1"
    id="side-nav"
    role="navigation"
  />
  <button
    aria-label="Toggle side navigation"
    className="emotion-2"
    id="toggle-sidenav"
    onClick={[Function]}
    title="Toggle SideNav"
  >
    Toggle
  </button>
</div>
`;

exports[`Sidebar renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  max-width: 1020px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-4"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <div
    className="emotion-3"
  >
    <aside
      className="emotion-1"
      id="primary-sidebar"
      role="complementary"
    />
    <main
      className="emotion-2"
      id="content"
      role="main"
    />
  </div>
</div>
`;

exports[`Topbar renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style: none;
  position: relative;
  z-index: 1000;
  padding: 0;
}

.emotion-0 a {
  background-color: var(--theme-ui-colors-bg_header,#fff);
  display: block;
  position: absolute;
  font-family: body;
  left: -9999px;
  padding: 1em;
}

.emotion-0 a:focus {
  left: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-bg_topbar,#355cac);
  display: block;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  overflow-x: scroll;
  white-space: nowrap;
  max-width: 1260px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-0"
  >
    <li>
      <a
        href="#content"
      >
        Skip to content
      </a>
    </li>
    <li>
      <a
        href="#footer"
      >
        Skip to footer
      </a>
    </li>
  </ul>
  <aside
    className="emotion-2"
    role="complementary"
  >
    <div
      className="emotion-1"
    >
      Test
    </div>
  </aside>
</div>
`;
